profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/domain/Lan.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.domain;
+
+import com.google.auto.value.AutoValue;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class Lan {
+
+   public abstract String id();
+
+   public abstract String type();
+
+   public abstract String href();
+
+   @Nullable
+   public abstract Metadata metadata();
+
+   @Nullable
+   public abstract Properties properties();
+
+   @Nullable
+   public abstract Entities entities();
```

Comment:
My knowledge is that, whether some field is present or not, is usually controlled by the depth param. So, as you can imagine, some fields may be existent, but if depth is too shallow they may be 'null'.

So what we've done in the depth testing, is to pull a response for each of the depth levels and simply try to instantiate a DataCenter object (which contains almost all the other domain objects). If we see the test failing (because of null entry), we put nullable on the field.

Now one may argue, that the field can still be null (in the server database), and it would be correct thing to say. To check this, we can go and try to instantiate each object with null on these fields or simply check in the UI if it can be empty.